File: web/src/api/drawing/project.js
import service from '@/utils/request'

export const CreateProject = (data) => {
    return service({
      url: '/drawing/project/CreateProject',
      method: 'post',
      data
    })
}

export const GetAllProject = (params) => {
  return service({
    url: '/drawing/project/GetAllProject',
    method: 'get',
    params
  })
}




//  画笔相关
export const GetAllBranch = (params) => {
    return service({
      url: '/drawing/project/GetAllBranch',
      method: 'get',
      params
    })
}
export const CreateBranch = (data) => {
    return service({
      url: '/drawing/project/CreateBranch',
      method: 'post',
      data
    })
}
export const DeleteBranch = (data) => {
    return service({
      url: '/drawing/project/DeleteBranch',
      method: 'post',
      data
    })
}

export const GetProjectById = (data) => {
  return service({
    url: '/drawing/project/GetProjectById',
    method: 'post',
    data
  })
}
export const GetParamsByProjectId = (data) => {
  return service({
    url: '/drawing/project/GetParamsByProjectId',
    method: 'post',
    data
  })
}
export const CreateParam = (data) => {
  return service({
    url: '/drawing/project/CreateParam',
    method: 'post',
    data
  })
}
export const ChangeParamBranch = (data) => {
  return service({
    url: '/drawing/project/ChangeParamBranch',
    method: 'post',
    data
  })
}
export const ChangeParam = (data) => {
  return service({
    url: '/drawing/project/ChangeParam',
    method: 'post',
    data
  })
}
export const DeleteParam = (data) => {
  return service({
    url: '/drawing/project/DeleteParam',
    method: 'post',
    data
  })
}
export const CreateImage = (data) => {
  return service({
    url: '/drawing/project/CreateImage',
    method: 'post',
    data
  })
}

File: server/router/drawing/project.go
package drawing

import (
	v1 "github.com/defeng-hub/ByOfficeAutomatic/server/api/v1"
	"github.com/gin-gonic/gin"
)

type ProjectRouter struct{}

func (ProjectRouter) Init(Router *gin.RouterGroup) {
	R := Router.Group("drawing/")
	Api := v1.ApiGroupApp.DrawingApiGroup.ProjectApi
	paramApi := v1.ApiGroupApp.DrawingApiGroup.ParamApi
	{
		R.GET("project/GetAllProject", Api.GetAllProject)
		R.POST("project/GetProjectById", Api.GetProjectById)
		R.POST("project/CreateProject", Api.CreateProject)
		R.POST("project/CreateBranch", Api.CreateBranch)
		R.POST("project/DeleteBranch", Api.DeleteBranch)
		R.GET("project/GetAllBranch", Api.GetAllBranch)
	}
	{
		// param
		R.POST("project/GetParamsByProjectId", paramApi.GetParamsByProjectId)
		R.POST("project/CreateParam", paramApi.CreateParam)
		R.POST("project/ChangeParamBranch", paramApi.ChangeParamBranch)
		R.POST("project/ChangeParam", paramApi.ChangeParam)
		R.POST("project/DeleteParam", paramApi.DeleteParam)
		R.POST("project/CreateImage", paramApi.CreateImage)
	}
}
